### PR TITLE
ARROW-15688: [C++] add_checked doesn't error out on duration overflow

### PR DESCRIPTION
**Pull request by rok**
This is to resolve [ARROW-15688](https://issues.apache.org/jira/browse/ARROW-15688).

### CODE REVIEW

**Comment by rok** [comment]
@pitrou thanks for spotting that one!

File: cpp/src/arrow/compute/kernels/scalar_temporal_test.cc [hunk 1]
```diff
@@ -28,13 +28,25 @@
...
+#include "arrow/util/string.h"
 
 namespace arrow {
 
 using internal::StringFormatter;
 
 namespace compute {
 
+template <typename... Elements>
+std::string MakeArray(Elements... elements) {
```
Comment (cyb70289) [inline cpp/src/arrow/compute/kernels/scalar_temporal_test.cc:40]: Looks this function is not very necessary. Can we directly write the test string at caller side?

Comment (rok) [reply cpp/src/arrow/compute/kernels/scalar_temporal_test.cc]: Done.

**Review by cyb70289 — APPROVED** [review]
+1

**Pull request closed without merging**